Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/VaccinationInfoCard.kt

```diff
@@ -22,21 +26,63 @@ class VaccinationInfoCard(parent: ViewGroup) :
     ) -> Unit = { item, payloads ->
         val curItem = payloads.filterIsInstance<Item>().singleOrNull() ?: item
         val daysUntilImmunity = curItem.daysUntilImmunity
-        body.text = when {
-            daysUntilImmunity == 1 -> context.resources.getString(
-                R.string.vaccination_list_immunity_tomorrow_card_body
-            )
-            daysUntilImmunity != null -> context.resources.getQuantityString(
-                R.plurals.vaccination_certificate_days_unti_immunity,
-                daysUntilImmunity,
-                daysUntilImmunity
-            )
-            else -> context.getString(R.string.vaccination_certificate_incomplete_vaccination)
+        val vaccinationStatus = curItem.vaccinationStatus
+        val boosterRule = curItem.boosterRule
+        val daysSinceLastVaccination = curItem.daysSinceLastVaccination
+        title.text = context.resources.getString(R.string.vaccination_state_title)
+        subtitle.text = context.resources.getString(
+            R.string.vaccination_days_since_last_shot,
+            daysSinceLastVaccination
+        )
+
+        when (vaccinationStatus) {
+            VaccinatedPerson.Status.COMPLETE -> {
+                body.text = when {
+                    daysUntilImmunity == 1 -> context.resources.getString(
+                        R.string.vaccination_list_immunity_tomorrow_card_body
+                    )
+                    daysUntilImmunity != null -> context.resources.getQuantityString(
+                        R.plurals.vaccination_certificate_days_unti_immunity,
+                        daysUntilImmunity,
+                        daysUntilImmunity
+                    )
+                    else -> context.getString(R.string.vaccination_certificate_incomplete_vaccination)
+                }
+            }
+
+            VaccinatedPerson.Status.BOOSTER_ELIGIBLE -> {
+
+                body.text = context.resources.getString(
+                    R.string.vaccination_card_booster_eligible,
+                    boosterRule!!.rule.identifier
+                )
+
+                body2Faq.isVisible = true
+                boosterBadge.isVisible = true
+                body2Faq.setTextWithUrl(
```

Comment:
it should be visible only when `vaccinatedPerson.hasBoosterNotification` is true,
i.e `boosterBadge.isVisible=vaccinatedPerson.hasBoosterNotification`

Reply:
good point! Will be added :)